node-notes-cli/app.js

```diff
@@ -0,0 +1,157 @@
+import { writeFile } from 'node:fs/promises';
+import { readFileSync } from 'node:fs';
+
+// Set up function and variables to handle user inputs
+const action = process.argv[2];
+const arg3 = process.argv[3];
+const arg4 = process.argv[4];
+const legalActions = ['read', 'create', 'update', 'delete'];
+
+function interpretInputs() {
+  switch (action) {
+    case 'read':
+      readNotes();
+      break;
+    case 'create':
+      newNote();
+      break;
+    case 'update':
+      updateNote();
+      break;
+    case 'delete':
+      deleteNote();
+      break;
+    default:
+      console.error(action, 'is not an acceptable action. Please try again with:', legalActions);
+      process.exit(1);
+  }
+}
+
+/**
+ * Reusable functions
+ */
+
+// Catch error handling
+function handleError(err) {
+  console.error(err);
+  process.exit(1);
+}
+
+// Evaluate ID
+async function evaluateId() {
+  try {
+    const notes = await getJournal('notes');
+    const validIds = Object.keys(notes);
+
+    if (typeof Number(arg3) !== 'number' || !validIds.includes(arg3)) {
+      console.error(arg3, 'is not a valid ID. Please provide a valid ID:', validIds.join(', '));
+      process.exit(1);
+    }
+  } catch (err) {
+    handleError();
+  }
+}
+
+// Multiple notes
+function tooManyNotes() {
+  console.error('Please note only one note at a time can be inputted. Try again and ensure your note is enclosed in quotation marks.');
+  process.exit(1);
+}
+
+// Journal retrieval
+async function getJournal(item) {
```

Comment:
I don't like using `get` on function names. The reason is that `get` has been overloaded in the industry to mean way too many things, so it's not clear what is happening here. Since you are reading the journal from the file system, a name like "readJournal" would be provide more information.